File: Storage/Code/389.找不同.py
#
# @lc app=leetcode.cn id=389 lang=python3
#
# [389] 找不同
#

# @lc code=start
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        if len(s)==0:
            return t
        dict={}
        for i in range(len(s)):
            if (s[i] in dict)==False:
                dict[s[i]]=1
            else:
                dict[s[i]]+=1
        for i in range(len(t)):
            if((t[i] in dict)==False or dict[t[i]]==0):
                return t[i]
            else:
                dict[t[i]]-=1
        return ""

# @lc code=end

File: Storage/Code/389.找不同.cpp
/*
 * @lc app=leetcode.cn id=389 lang=cpp
 *
 * [389] 找不同
 */

// @lc code=start
class Solution {
public:
    char findTheDifference(string s, string t) {
        vector<int> vec(26,0);
        for(auto &c: s){
            ++vec[c-'a'];
        }
        for(auto &c: t){
            if(vec[c-'a']==0){
                return c;
            }
            --vec[c-'a'];
        }
        return ' ';
    }
};
// @lc code=end

